array.c

```diff
@@ -4220,6 +4222,66 @@ rb_ary_diff(VALUE ary1, VALUE ary2)
     return ary3;
 }
 
+
+/*
+ *  call-seq:
+ *     ary.difference(other_ary1, other_ary2,...)   -> ary
+ *
+ *  Array Difference
+ *
+ *  Returns a new array that is a copy of +self+, removing any items
+ *  that also appear in any of the +other_ary+s. The order of +self+ is
+ *  preserved.
+ *
+ *  It compares elements using their #hash and #eql? methods for efficiency.
+ *
+ *     [ 1, 1, 2, 2, 3, 3, 4, 5 ].difference([ 1, 2, 4 ])       #=>  [ 3, 3, 5 ]
+ *     [ 1, 'c', :s, 'yep' ].difference([ 1 ], [ 'a', 'c' ])    #=> [:s, "yep"]
+ *
+ *  If you need set-like behavior, see the library class Set.
+ *
+ *  See Array#-.
+ */
+
+static VALUE
+rb_ary_difference_multi(int argc, VALUE *argv, VALUE ary)
+{
+    VALUE ary_diff;
+    bool *is_hash;
+    long i, length;
+
+    length = RARRAY_LEN(ary);
+    is_hash = (bool *)malloc(sizeof(bool)*argc);
```

Comment:
`malloc` may return `NULL` and it can leak when the following comparisons raise, use `ALLOCV_N` / `ALLOCV_END` instead.

Reply:
And `is_hash` isn't `free`d.